### PR TITLE
Issue 5196: REST call for fetching readergroup properties does not work when TLS is enabled in the standalone

### LINKED ISSUE #5196: REST call for fetching readergroup properties does not work when TLS is enabled in the standalone
**Problem description**
Making the rest call using the uri `/v1/scopes/{scope}/readergroups/{readergroup}` fails when done against pravega-standalone with TLS enabled. The error code returned is `500`. The same call succeeds when TLS is disabled (TLS disabled with authentication/authorization enabled also works). 

**Problem location**
Controller REST API

**Suggestions for an improvement**
Check `StreamMetadataResourceImpl` for the codeflow with TLS enabled.

### PR DESCRIPTION
**Pull request by anirudhkovuru**
**Change log description**  
The fix allows for readergroup properties to be queried through the REST API when TLS is enabled.

**Purpose of the change**  
Fixes #5196 

**What the code does**
- The class `StreamMetadataResourceImpl` now has a new field for `ClientConfig` which is to be passed through its constructor.
- When generating a `ClientFactory` in order to get the readergroup properties, the above `ClientConfig` is passed. 

**How to verify it**  
All tests should pass and build should succeed.

### CODE REVIEW

File: controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java [hunk 1]
```diff
@@ -196,7 +196,7 @@ public void setup() throws Exception {
                                                                         .controllerURI(URI.create("tcp://localhost"))
                                                                         .build());
         restServer = new RESTServer(controller, mockControllerService, authManager, serverConfig,
-                connectionFactory);
+                connectionFactory, ClientConfig.builder().controllerURI(URI.create("tcp://localhost")).build());
```
Comment (shiveshr) [inline controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java:199]: nit: you dont need to give controller uri here for this test..

**Comment by shiveshr** [comment]
Code coverage is 75%. please add unit tests covering the changes.

File: controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java (hunk 1 above)
Comment (anirudhkovuru) [reply controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java]: removed it

File: controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataAuthFocusedTests.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ public static void initializer() throws IOException, InvalidKeySpecException, No
                                                                   .controllerURI(URI.create("tcp://localhost"))
                                                                   .build());
         restServer = new RESTServer(controller, mockControllerService, authManager, serverConfig,
-                connectionFactory);
+                connectionFactory, ClientConfig.builder().controllerURI(URI.create("tcp://localhost")).build());
```
[the remark this answers is not in the document]
Comment (anirudhkovuru) [reply controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataAuthFocusedTests.java]: removed it

**Review by shiveshr — APPROVED** [review]

**Pull request merged**